src/NServiceBus.Core/Pipeline/Incoming/LoadHandlersConnector.cs

```diff
@@ -57,8 +66,29 @@ async Task<CompletableSynchronizedStorageSession> AdaptOrOpenNewSynchronizedStor
                    ?? await synchronizedStorage.OpenSession(contextBag).ConfigureAwait(false);
         }
 
+        static void LogHandlersInvocation(IIncomingLogicalMessageContext context, List<MessageHandler> handlersToInvoke)
+        {
+            var builder = new StringBuilder();
+            builder.AppendLine($"Processing message type: {context.Message.MessageType}");
+            builder.AppendLine($"Headers: {string.Join(", ", context.Headers.Select(h => $"{h.Key}:{h.Value}").ToArray())}");
+            builder.AppendLine($"Message: {context.Message.Instance}");
+            builder.Append("Handlers to invoke: ");
+
+            foreach (var messageHandler in handlersToInvoke)
+            {
+                builder.Append($"{messageHandler.HandlerType.FullName} => ");
+            }
+
+            builder.Length -= 4;
```

Comment:
What does this do?

Reply:
Getting rid of the last trailing ` => ` that is automatically appened.